From be18ae72161414a9f321376d8e6e525d1b0063b1 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Alex=20R=C3=B8nne=20Petersen?= <alex@alexrp.com>
Date: Wed, 13 Aug 2025 04:48:51 +0200
Subject: [PATCH 081/133] std.hash.xxhash: disable xxhash3 test on all mips64
 targets
X-Developer-Signature: v=1; a=openpgp-sha256; l=1757; i=xtex@astrafall.org;
 h=from:subject; bh=OwNCwaspR7m3JVlU8UMrwpfzQAbB5CElaTvQkpgdRik=;
 b=owGbwMvMwCW2U4Ij7wZL9ETG02pJDJn78tjWTGJOqH536cjSIPnEXxd9LM6IplXft5vjGG02a
 5f3lCMWHRNZGMS4GCzFFFmKDBu8WXXS+UWXlcvCzGFlAhkiLdLAwMDAwMLAl5uYV2qkY6Rnqm2o
 Z2ioY6RjxMDFKQBTHanC8D+Dabbwvr2mEamXbZKWBocvKfy0g4/jmL3361Qj7oiph8wY/mn4zpg
 TuK4joz3h0m+u/R+TmNqLd1itCtxneaIhWMLQhgsA
X-Developer-Key: i=xtex@astrafall.org; a=openpgp;
 fpr=7231804B052C670F15A6771DB918086ED8045B91

See also dd1de18f9629904b5d97d5791a47244ac63385da.

https://github.com/ziglang/zig/issues/23807
(cherry picked from commit 59ee4e8e5484a2f976485fca5b085a7ba0358074)
---
 lib/std/hash/xxhash.zig | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/lib/std/hash/xxhash.zig b/lib/std/hash/xxhash.zig
index b3128f39b2cd..270759802e29 100644
--- a/lib/std/hash/xxhash.zig
+++ b/lib/std/hash/xxhash.zig
@@ -780,7 +780,7 @@ fn testExpect(comptime H: type, seed: anytype, input: []const u8, expected: u64)
 }
 
 test "xxhash3" {
-    if (builtin.cpu.arch.isMIPS64() and (builtin.abi == .gnuabin32 or builtin.abi == .muslabin32)) return error.SkipZigTest; // https://github.com/ziglang/zig/issues/23807
+    if (builtin.cpu.arch.isMIPS64()) return error.SkipZigTest; // https://github.com/ziglang/zig/issues/23807
 
     const H = XxHash3;
     // Non-Seeded Tests
@@ -813,7 +813,7 @@ test "xxhash3" {
 }
 
 test "xxhash3 smhasher" {
-    if (builtin.cpu.arch.isMIPS64() and (builtin.abi == .gnuabin32 or builtin.abi == .muslabin32)) return error.SkipZigTest; // https://github.com/ziglang/zig/issues/23807
+    if (builtin.cpu.arch.isMIPS64()) return error.SkipZigTest; // https://github.com/ziglang/zig/issues/23807
 
     const Test = struct {
         fn do() !void {
@@ -826,7 +826,7 @@ test "xxhash3 smhasher" {
 }
 
 test "xxhash3 iterative api" {
-    if (builtin.cpu.arch.isMIPS64() and (builtin.abi == .gnuabin32 or builtin.abi == .muslabin32)) return error.SkipZigTest; // https://github.com/ziglang/zig/issues/23807
+    if (builtin.cpu.arch.isMIPS64()) return error.SkipZigTest; // https://github.com/ziglang/zig/issues/23807
 
     const Test = struct {
         fn do() !void {
-- 
2.52.0

